plugins/modules/postgresql_subscription.py

```diff
@@ -732,13 +739,20 @@ def main():
                                           check_mode=module.check_mode)
 
         else:
-            if connparams:
-                connparams = cast_connparams(connparams)
-
-            changed = subscription.update(connparams,
-                                          publications,
-                                          subsparams,
-                                          check_mode=module.check_mode)
+            if subscription.attrs['conninfo'] != {}:
+                if connparams:
+                    connparams = cast_connparams(connparams)
+
+                changed = subscription.update(connparams,
+                                              publications,
+                                              subsparams,
+                                              check_mode=module.check_mode)
+            else:
+                module.warn("'connparams' is ignored when existing subscription connparms are unknowable")
```

Comment:
```suggestion
                module.warn("'connparams' is ignored because it's not supported by the database engine.")
```
Something like that?

Reply:
That's better, but this could be confusing about it being an issue specific to the implementation/hosting of Postgres, rather than an issue inherent to Postgres itself.

Perhaps we could just say what the actual issue is? Something like

`'connparams' is ignored because pg_subscription.subconninfo is not accessible in your instance.`

Either way, we will want to be very explicit about it in the documentation that will correspond with this. I'll work on that and the changelog fragment today.